components/mosart/src/riverroute/RtmMod.F90

```diff
@@ -3515,63 +3547,12 @@ subroutine MOSART_init
           TUnit%wr_bf = 0.0_r8   
 
           allocate( TUnit%e_eprof_in2( 11, begr:endr ) )    
```

Comment:
Memory allocation here is assuming that there will be at most 11 layers. But, in [`read_elevation_profile`](https://github.com/E3SM-Project/E3SM/pull/3754/files#diff-659dea48e8aa2bc7dd7a44a748a8e357R4083) the number of layers could greater than 11. Thus, move the memory allocation of `TUnit%e_eprof_in2` within `read_elevation_profile`.

Reply:
There is pre-defined number of elevation profile: Tctl%npt_elevProf = 11, which is read from mosart_in. It is also used to allocate other elevation profile related variable memory. 

How about we add a check in the subroutine read_elevation_profile that if the number of points in the elevation profile is not consistent with Tctl%npt_elevProf then the program stop.